langchain/retrievers/contextual_compression.py

```diff
@@ -0,0 +1,111 @@
+""""""
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Dict, List, Optional
+
+from pydantic import BaseModel, Extra, Field, root_validator
+
+from langchain import BasePromptTemplate, LLMChain, PromptTemplate
+from langchain.schema import BaseLanguageModel, BaseRetriever, Document
+
+
+def default_get_input(query: str, doc: Document) -> Dict[str, Any]:
+    """Return the compression chain input."""
+    return {"question": query, "context": doc.page_content}
+
+
+class BaseDocumentFilter(BaseModel, ABC):
```

Comment:
to me filter sounds like it's just choosing which docs to keep. maybe BaseDocumentCompressor/Compression?